E-Commerce/readme.md

```diff
@@ -46,13 +46,73 @@ erDiagram
 Write SQL queries to complete the following tasks:
 
 - [ ] List all the products whose name contains the word "socks"
+
+```sql
+select * from products where product_name ilike '%socks%';
+```
+
 - [ ] List all the products which cost more than 100 showing product id, name, unit price, and supplier id
+
+```sql
+select p.id, p.product_name, pa.unit_price, pa.supp_id
+from products p  INNER JOIN
+product_availability pa on (p.id = pa.prod_id)
+where pa.unit_price > 100
+order by unit_price desc;
+```
+
 - [ ] List the 5 most expensive products
+
+```sql
+select p.id, p.product_name, pa.unit_price
+from products p  INNER JOIN
+product_availability pa on (p.id = pa.prod_id)
+order by unit_price desc limit 5;
+```
+
 - [ ] List all the products sold by suppliers based in the United Kingdom. The result should only contain the columns product_name and supplier_name
+
+
+```sql
+select p.product_name, s.supplier_name
+from suppliers s  LEFT JOIN
+product_availability pa on (s.id = pa.supp_id)
+LEFT JOIN products p on (p.id = pa.prod_id)
+where s.country = 'United Kingdom'
+order by pa.unit_price desc;
+```
```

Comment:
Whilst this is essentially correct, be careful about the use of LEFT JOIN instead of INNER JOIN, depending upon the output you require.  The question here suggests we are interested in products, whereas suppliers here who have no product_availability or product may still yield rows in this output.  (See https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-joins/ for clarification.)